common/types/activation.go

```diff
@@ -78,7 +101,7 @@ func (t *ATXID) UnmarshalText(buf []byte) error {
 }
 
 // EmptyATXID is a canonical empty ATXID.
-var EmptyATXID = ATXID{}
+var EmptyATXID = &ATXID{}
```

Comment:
When we switch to pointer semantics, comparison against this value become useless since you actually need to deref the pointer to check against it, however, when using pointer semantics you actually _can't_ compare against this because you're basically just [comparing memory addresses](https://go.dev/play/p/hI98vzXVXjr). If we want to use pointer semantics I would actually scratch this empty value and use `nil`'s wherever necessary. It isn't really clear what does an "Empty ATX ID" even mean. I'd personally use a `nil` there (I know about the `billion dollar mistake`, however, we don't have Rust's typesystem here unfortunately).